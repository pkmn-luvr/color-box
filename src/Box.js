import React from 'react';

function Box({ id, width, height, backgroundColor, removeBox }) {
  return (
    <div>
      <div
        style={{
          width: `${width}px`,
          height: `${height}px`,
          backgroundColor
        }}
      />
      <button onClick={removeBox}>X</button>
    </div>
  );
}

export default Box;
