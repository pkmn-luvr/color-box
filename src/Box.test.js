import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Box from './Box';

// Smoke Test
test('renders Box component without crashing', () => {
  render(<Box id="1" width="100" height="100" backgroundColor="pink" removeBox={() => {}} />);
});

// Snapshot Test
test('Box component matches snapshot', () => {
  const { asFragment } = render(<Box id="1" width="100" height="100" backgroundColor="pink" removeBox={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});
