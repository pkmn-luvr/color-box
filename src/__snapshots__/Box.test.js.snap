// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      style="width: 100px; height: 100px; background-color: pink;"
    />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
